src/devices/GrovePi/Sensors/AnalogSensor.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Iot.Device.GrovePiDevice.Models;
+
+namespace Iot.Device.GrovePiDevice.Sensors
+{
+    /// <summary>
+    /// AnalogSensor is a generic analog sensor
+    /// </summary>
+    public class AnalogSensor : ISensor<int>
+    {
+        internal GrovePi _grovePi;
+
+        /// <summary>
+        /// On GrovePi, ADC is 1023
+        /// </summary>
+        public int MaxAdc => 1023;
+
+        /// <summary>
+        /// AnalogSensor constructor
+        /// </summary>
+        /// <param name="grovePi">The GrovePi class</param>
+        /// <param name="port">The grove Port, need to be in the list of SupportedPorts</param>
+        public AnalogSensor(GrovePi grovePi, GrovePort port)
+        {
+            if (!SupportedPorts.Contains(port))
+                throw new ArgumentException($"Error: grove Port not supported");
+            _grovePi = grovePi;
+            Port = port;
+            _grovePi.PinMode(Port, PinMode.Input);
```

Comment:
I feel like GrovePi could probably implement IGpioController - might be worth taking a look

Reply:
It actually can but the IGpioController is only for Digital Pins and not for analogic pins. And most people who are using Grove sensors which are not implemented will look at the Arduino code. And in the Arduino code, they'll find analogRead, digitalWrite, etc. And not the ones from the IGpioController. So in this case, I think it will bring more confusion than Clarity. If I would implement the IGpioController, I would anyway keep the DigitalWrite and DigitalRead functions.
GrovePi is here to bring analogic sensors to the RaspberyPi world and makes things easy. So I'll keep the philosophie of GrovePi.

Reply:
@Ellerbach I'd suggest to create an issue that we should have support for analog pins in Gpio

Reply:
I will do. also the IGpioInterface is not present in the nuget 0.1.0-prerelease.19078.2. Would be great to have a new nuget release that would include it.

Reply:
Issue for analogic support: https://github.com/dotnet/iot/issues/255